### DISCUSSION TITLE
Is there a focused monitor indicator?

### DISCUSSION THREAD

**Discussion by rickswe** [Alexays/Waybar#3084] [Q&A]
Hello, I'm using river.

Is there a way to see what monitor I'm on, perhaps an indicator icon or something I can use?

When I have no windows open it's hard for me to know what monitor I'm on. When I have
windows open it's obvious because I have the border highlighted by my window manager.

Thanks

**Comment by ordy** [comment]
`#workspaces button.current_output.focused`

**Reply by rickswe** [reply]
Hello, thanks for the suggestion. This is not working, are there some additional things I need to add?

style.css
```css
#workspaces button.current_output.focused{
  background: #2AA198;
  color: #ffffff;
}
```

config

```json
 "river/tags": {
        "num-tags":6,
        "tag-labels":[
      "a",
      "b",
      "c",
    ],
    "disable-click": false
    },
  "river/window": {
    "format": " {} ",
    "max-length": 98,
    "tooltip": true
  },

  "river/mode": {
    "format": "mode: {}"
  },
  "tray": {
    "icon-size": 20,
    "spacing": 2
  },

```

**Reply by ordy** [reply]
Oh, you are on River. Not familiar with river/tags, but I don't think they have a current_output styling class. (at least it's not mentioned in the documentation).

**Reply by rickswe** [reply]
UPDATE: I figured it out, see comment below. If there is something much simpler please let me know.

I think the work around would be to create a custom module in the center that would ouput
an emoji based on a keyboard shortcut trigger. I'm not too familiar with custom modules but:

It would have to be trigered from a script.

```sh
riverctrl some-key-combo my-river-helper.sh next-screen

# my-river-helper.sh
riverctrl focus-output next 
pkill -SIGRTMIN+9 waybar
````

test.sh

```sh
#!/bin/bash

pkill -SIGRTMIN+9 waybar

echo "{\"text\": \"EMOJI\"}"

```

config

```json
"modules-center": ["custom/test"],

"custom/test": {
    "exec": "~/test.sh $RANDOM",
    "on-click-right": "~/test.sh",
    "return-type": "json",
    "exec-if": "exit 0",
    "interval": 5000,
    "format": {},
    "signal": 9
},
```

I would have to figure out how to output only on the monitor that is in focused.
I prefer not to have an interval, instead it be triggered by a keyboard shortcut.
Honestly I'm not familiar with custom modules. Can you help?

Thanks

**Comment by rickswe** [comment]
I did figure it out in a round about way by creating custom modules. However, if there is
a much simpler solution please comment. I will put the explanation in terms of river
because that is what I use.

You must have two different custom modules for each monitor, one will
exec with argument '1' and the other with '2' with the same signal.
they both execute the same script return-my-icon.sh.

1. Your keyboard shortcut in init should trigger a script like river.sh next_monitor
2. river.sh this will modify with sed a temp file to keep track of the current monitor.
it will then execute `pkill -SIGRTMIN+8 waybar` which will trigger return-my-icon.sh
3. return-my-icon.sh then checks what the current monitor is and return an icon or no icon.
4. You can also return json text and assign a class that will dim your clock or icon to indicate what monitor you are on.

**Comment by rickswe** [comment]
Update: All though this works well it does not account for when the mouse is moved.
In river there is no way to trigger an event when the mouse moves to the next monitor.
Therefore, I will have to use waybar if there is such an option to style or hide the
unfocused bar:

Something like this:

```css
window#waybar.notfocused #window {  
 background:none;  
}  
```
